### PR TITLE
Merge branch

### PR DESCRIPTION
**Pull request by paramsingh**
merge latest production changes into master.

### CODE REVIEW

**Comment by paramsingh** [comment]
Also added tests for the new case-insensitive feature in this PR.

File: messybrainz/db/data.py [hunk 1]
```diff
@@ -144,12 +144,14 @@ def submit_recording(connection, data):
     Returns:
         the Recording MessyBrainz ID of the data
     """
-    data_json = json.dumps(data, sort_keys=True, separators=(',', ':'))
-    data_sha256 = sha256(data_json.encode("utf-8")).hexdigest()
+    sha256_json = convert_to_messybrainz_json(data, transform=True)
```
Comment (mayhem) [inline messybrainz/db/data.py]: GAH! We're doing the serialization twice! Would it make sense to do it once and also kick out the lower case sha value as a side-effect that can be ignored?

**Review by mayhem — COMMENTED** [review]
I'm torn between fixing this now and allow a bug for a future enhancement to be entered and proceeding. You decide and proceed!

**Review by mayhem — APPROVED** [review]
🍔 🐹 🔨

**Pull request merged**